ui/qt-lib/src/Ariadne/UI/Qt/Widgets/WalletTree.hs

```diff
@@ -81,17 +83,27 @@ initWalletTree langFace itemModel selectionModel = do
   return (widget, WalletTree{..})
 
 addWalletClicked :: UiLangFace -> WalletTree -> Bool -> IO ()
-addWalletClicked UiLangFace{..} WalletTree{..} _checked =
+addWalletClicked langFace@UiLangFace{..} walletTree@WalletTree{..} _checked =
   runNewWallet >>= \case
     NewWalletCanceled -> return ()
-    NewWalletAccepted NewWalletParameters{..} -> void $ langPutUiCommand $
+    NewWalletAccepted NewWalletParameters{..} ->
       case nwSpecifier of
-        NewWalletName -> UiNewWallet nwName nwPassword
-        NewWalletMnemonic mnemonic full -> UiRestoreWallet nwName nwPassword mnemonic full
+        NewWalletName -> doCreateNewWallet langFace walletTree nwName nwPassword
+        NewWalletMnemonic mnemonic full -> void $ langPutUiCommand $
+          UiRestoreWallet nwName nwPassword mnemonic full
+
+doCreateNewWallet :: UiLangFace -> WalletTree -> Text -> Maybe Text -> IO ()
+doCreateNewWallet UiLangFace{..} WalletTree{..} name password = do
+  let entropySize = 16 -- FIXME Hardcoded for now, will have to fetch from wallet config
+  mnemonic <- mnemonicGenerator entropySize
+  confirmationResult <- runConfirmMnemonic mnemonic
+
+  case confirmationResult of
+    ConfirmMnemonicSuccess -> void $ langPutUiCommand $ UiRestoreWallet name password (unwords mnemonic) False
+    ConfirmMnemonicFailure -> return ()
```

Comment:
Do we want some message box in this case?

Reply:
Can you ask in #ariadne-design please?

Reply:
Asked.